drizzlepac/alignimages.py

```diff
@@ -82,45 +82,72 @@ def check_and_get_data(input_list,**pars):
-    for input_item in input_list:
-        try:
-            filelist = aqutils.retrieve_observation(input_item,**pars)
-        except Exception:
-            filelist = []
-        if len(filelist) == 0:
-            # look for local copy of the file
-            input_rootname = input_item[:input_item.find('_')]
-            fitsfilenames = sorted(glob.glob("{}_fl?.fits".format(input_rootname)))
-            if len(fitsfilenames) > 0:
-                imghdu = fits.open(fitsfilenames[0])
-                imgprimaryheader = imghdu[0].header
-                try:
-                    asnid = imgprimaryheader['ASN_ID'].strip().lower()
-                    if asnid == 'NONE':
-                        asnid = None
-                except KeyError:
-                    asnid = None
-                if asnid and asnid not in retrieve_list:
+
+    # If the regression test flag is True, a regression test has been invoked where the input_list 
+    # actually contains the full filename(s) (e.g., *_flt.fits).  Note that not all regression tests use 
+    # this functionality as Artifactory or a local disk is accessed for the data source with the use of 
+    # this flag.  Some regression tests use astroquery to connect to MAST. This regression testing code 
+    # is not a scalable solution, but it is necessary for the alignment portion of the drizzle software.
+    if pars['regtest'] == True:
+        # Test folder containing the data - the root path is defined by the environment variable TEST_BIGDATA
+        data_path = pars['test_data_path']
+        from ci_watson.artifactory_helpers import get_bigdata
+        for infile in input_list:
+            full_file_path = get_bigdata(data_path[0],data_path[1],data_path[2],infile)
+            totalInputList.append(infile)
+    else:
+        archive = pars['archive']
+        clobber = pars['clobber']
+        retrieve_list = [] # list of already retrieved ASNs, only request each once
+        for input_item in input_list:
+            try:
+                filelist = aqutils.retrieve_observation(input_item,archive=archive,clobber=clobber)
+            except Exception:
+                filelist = []
+            if len(filelist) == 0:
+                # look for local copy of the file
+                input_rootname = input_item[:input_item.find('_')]
+                fitsfilenames = sorted(glob.glob("{}_fl?.fits".format(input_rootname)))
+                if len(fitsfilenames) > 0:
+                    imghdu = fits.open(fitsfilenames[0])
+                    imgprimaryheader = imghdu[0].header
                     try:
-                        filelist = aqutils.retrieve_observation(asnid,**pars)
-                        retrieve_list.append(asnid) # record asn's already retrieved
-                    except Exception:
-                        filelist = []
+                        asnid = imgprimaryheader['ASN_ID'].strip().lower()
+                        if asnid == 'NONE':
```

Comment:
Will never be `NONE` due to `.lower()`

Reply:
Also see https://github.com/spacetelescope/drizzlepac/pull/283#issuecomment-476452356

Reply:
Changed 'NONE' to 'none'.  I need to address bugs/errors in this PR.  Better coding practices and efficiency improvements will be handled in a different Issue which references this PR.

Reply:
The code in this routine is being entirely reworked.